cmake/modules/boards.cmake

```diff
@@ -164,3 +172,23 @@ if(NOT BOARD_DIR)
 endif()
 
 add_custom_target(boards ${list_boards_commands} USES_TERMINAL)
+
+# Board extensions are enabled by default
+set(BOARD_EXTENSIONS ON CACHE BOOL "Support board extensions")
+
+# Process board extensions
+if(BOARD_EXTENSIONS)
+  get_filename_component(board_dir_name ${BOARD_DIR} NAME)
+
+  foreach(root ${BOARD_ROOT})
+    set(board_extension_dir ${root}/boards/extensions/${board_dir_name})
+    if(NOT EXISTS ${board_extension_dir})
+      continue()
+    endif()
+
+    zephyr_file(CONF_FILES ${board_extension_dir}
+      DTS   board_extension_dts_files
+      KCONF board_extension_conf_files
+    )
```

Comment:
not convinced this belongs here.

The `boards.cmake` file is purely for board handling and not related to dts or Kconfig.

Kconfig, even default board settings, are handled in `boards.cmake`.
Dts overlays, even board dts and revision overlays, are handled in `dts.cmake`.

Reply:
note: done because of this comment: https://github.com/zephyrproject-rtos/zephyr/pull/57390#discussion_r1184742073

Reply:
> note: done because of this comment: [#57390 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/57390#discussion_r1184742073)

yep, but that was referring to the lookup of the board extensions dir, just like we look up the board dir in `boards.cmake`.
So boards.cmake can set the exact board extension dir.

the intention with the rest of the comment
> ... and then the corresponding dts and kconfig modules decides how to use such information.

was to indicate that the use of the extension folder was to have the dts and kconfig decide how to use the information / what files to apply and how.

Reply:
So then, can you provide a concrete suggestion? To me the initial separate file was better/more clear.

Reply:
done, I hope I've done it right.